pkg/services/librarypanels/database.go

```diff
@@ -104,3 +108,54 @@ func (lps *LibraryPanelService) getAllLibraryPanels(c *models.ReqContext) ([]Lib
 
 	return libraryPanels, err
 }
+
+// updateLibraryPanel updates a Library Panel.
+func (lps *LibraryPanelService) updateLibraryPanel(c *models.ReqContext, cmd updateLibraryPanelCommand) (LibraryPanel, error) {
+	var libraryPanel LibraryPanel
+	err := lps.SQLStore.WithTransactionalDbSession(context.Background(), func(session *sqlstore.DBSession) error {
+		panelInDb, err := getLibraryPanel(session, cmd.UID, c.SignedInUser.OrgId)
+		if err != nil {
+			return err
+		}
+
+		libraryPanel = LibraryPanel{
+			ID:        panelInDb.ID,
+			OrgID:     c.SignedInUser.OrgId,
+			FolderID:  cmd.FolderID,
+			UID:       cmd.UID,
+			Name:      cmd.Name,
+			Model:     cmd.Model,
+			Created:   panelInDb.Created,
+			CreatedBy: panelInDb.CreatedBy,
+			Updated:   time.Now(),
+			UpdatedBy: c.SignedInUser.UserId,
+		}
+
+		if cmd.FolderID == 0 {
```

Comment:
You shouldn't _conditionally_ update a resource when handling a PUT request, the semantics of PUT in REST is to update the _whole_ resource. If you want partial updates, you should use PATCH.

Reply:
Oh man, sorry I messed up 🙈 . I'll fix it in the next commit.

Reply:
Fixed